src/amazonica/aws/dynamodbv2.clj

```diff
@@ -38,7 +38,7 @@
 (defn- scalar-attribute-value? [value]
   (and (map? value)
        (= 1 (count value))
-       (= 1 (count (name-of-first-key value)))))
+       (contains? #{:s :b :BOOL :SS :BS :n :NS :l :m :NULL} (name-of-first-key value))))
```

Comment:
Good catch! I tested this with real Dynamo, and it worked, but you're
absolutely right that name-of-first-key returns a string.

I'll fix this and send a new pull request. If you tell me how to configure
Amazonica to use Dynamo Local, I'll be sure to test with that as well.

Please send me a code snippet that demonstrates this bug so I can be
completely sure.

On Saturday, 14 February 2015, Gordon Stratton <notifications@github.com>
wrote:

> In src/amazonica/aws/dynamodbv2.clj
> <https://github.com/mcohen01/amazonica/pull/121#discussion_r24705465>:
>
> > @@ -38,7 +38,7 @@
> >  (defn- scalar-attribute-value? [value]
> >    (and (map? value)
> >         (= 1 (count value))
> > -       (= 1 (count (name-of-first-key value)))))
> > +       (contains? #{:s :b :BOOL :SS :BS :n :NS :l :m :NULL} (name-of-first-key value))))
>
> @jmglov <https://github.com/jmglov> Is this correct? (name-of-first-key
> value) should return a string, so it seems like the contains? check would
> never be true. I'm getting the following dynamodb-local error, new in
> amazonica 0.3.14:
>
> com.amazonaws.AmazonServiceException: Invalid attribute value type (Service: AmazonDynamoDBv2; Status Code: 400; Error Code: ValidationException; Request ID: 022030b9-6264-4174-a258-cd02d4ced32f)
>
> and I believe this change is related. I can file a new issue but I thought
> I'd check first.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mcohen01/amazonica/pull/121/files#r24705465>.
>


-- 
Sent from my iPhone. Please excuse the brevity and potential typos.